src/cloud-api-adaptor/pkg/securecomms/apic/apic.go

```diff
@@ -0,0 +1,65 @@
+package apic
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/securecomms/sshutil"
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/util/netops"
+)
+
+var logger = sshutil.Logger
+
+type ApiClient struct {
+	servicePort uint16
+	nsPath      string
+}
+
+// Create a client for API_SERVER_REST service
+func NewApiClient(servicePort uint16, nsPath string) *ApiClient {
+	return &ApiClient{
+		servicePort: servicePort,
+		nsPath:      nsPath,
+	}
+}
+
+// GetKey uses kbs-client to obtain keys such as pp-sid/privateKey, sshclient/publicKey
+func (c *ApiClient) GetKey(key string) (data []byte, err error) {
+	url := fmt.Sprintf("http://127.0.0.1:%d/cdh/resource/default/%s", c.servicePort, key)
+
+	client := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (conn net.Conn, err error) {
```

Comment:
It would be good to add a comment here explaining why this method is overridden for people looking at the code in future